### DISCUSSION TITLE
Is it possible add atlantis style workflow (Apply before merge) ?

### DISCUSSION THREAD

**Discussion by kishaningithub** [dflook/terraform-github-actions#237] [Q&A]
## Context
- https://github.com/dflook/terraform-github-actions/discussions/67
- https://github.com/runatlantis/atlantis/discussions/2994

## Question
What do you think are the caveats if this action should support the "Apply before merge" workflow like atlantis?

**Comment by dflook** [comment]
I'm not familiar with atlantis, but the `dflook/terraform-apply` action can run anytime. The PR doesn't need to be merged.

You can run the `dflook/terraform-apply` job from `issue_comment` events. There is an example [here](https://github.com/dflook/terraform-github-actions/tree/main/terraform-apply#applying-a-plan-using-a-comment) of running the apply when someone comments `terraform apply` on a PR.

If you want to enforce who can run the apply and when it can be applied, you would need to implement that yourself. I think others have had problems ensuring the PR has had the required reviews.
